src/lib/metaphysics.coffee

```diff
@@ -3,6 +3,19 @@ qs = require 'qs'
 request = require 'superagent'
 { extend, some } = require 'underscore'
 { METAPHYSICS_ENDPOINT, API_REQUEST_TIMEOUT, REQUEST_ID } = require('sharify').data
+ip = require 'ip'
+
+resolveIPv4 = (ipAddress) ->
+  if ip.isV6Format(ipAddress)? and ~ipAddress.indexOf('::ffff')
+    return ipAddress.split('::ffff:')[1]
+  return ipAddress
+
+resolveProxies = (req) ->
+  ipAddress = resolveIPv4(req.connection.remoteAddress)
```

Comment:
So `remoteAddress` is the user's or the ELB's IP? It looks like it's being added to the end of the `X-Forwarded-For` chain (whereas the user's IP should be in the beginning, right?).

I'm prob still a tad confused, I would have expected that you either pass along the `X-Forwarded-For` headers as-is (since the Force ELB has already moved the client IP to the first, and appended itself).

or, if that's not present (for whatever reason), you can just pass along the `remoteAddress`.

Reply:
Ok yea a bit confusing.  From [the documentation](https://expressjs.com/en/guide/behind-proxies.html), `req.remoteAddress` is the "raw" address of the connection, so could be either the client IP (if the app is directly facing the internet) or the load balancer IP.  ( `req.ip` on the other hand, checks the `X-Forwarded-For` header for the first IP, so it's what you'd want to get the "real" client IP. )

The reason I am using this approach is that the Force ELB, as you say has already set the client IP in the `X-Forwarded-For` header but has _not_ appended its own IP.  That's why the branching statement in `resolveProxies` does - either appends the ELB ip if the header is set, or sets it if unset.

Reply:
I see, so if there's an `X-Forwarded-For` header, then the IP is assumed to be the ELB's, and it's appended and the whole chain passed through.

If that header isn't present, then (and this shouldn't really happen with our Force deployment tho right?) the IP is assumed to be the client and is just passed as-is.

That makes sense, thanks for the explanation 👍